### PR TITLE
Exposure Parsing

### PR DESCRIPTION
**Pull request by z3z1ma**
# Description

Parsing exposures from Metabase automatically in relation to a dbt project. This function is huge for understanding exposures within your data model. Close the loop between model composition and tangible BI exposures of those models creating an extremely valuable company asset in the dbt docs. In the nature of this project, the aim is we can use almost the same invocation parameters users set up in CI / CD for `dbt-metabase export` as they can with `dbt-metabase exposures`. This lets you run the invocation in production to essentially keep you docs in full sync with BI.

Note: Just opening this for visibility and preliminary code introspection. A bit more to do (but not too much). 

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

Open to ideas but ultimately I think making a mock api from metabase running on top of a database that has been seeded and ran through with the dbt jaffle shop project would be most robust. Just a lot of work. 

- [ ] CI test for exposure documentation

**Test Configuration**:
* Python version: 3.8

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

References #22

### CODE REVIEW

File: dbtmetabase/__init__.py [hunk 1]
```diff
@@ -69,9 +85,6 @@ def export(
         assert (
             schema and not schemas_excludes
         ), "Must target a single schema if using yaml parser, multiple schemas not supported."
-    assert bool(schema) != bool(
```
Comment (fernandobrito) [inline dbtmetabase/__init__.py:72]: This is actually fixing a bug that is available on existing `master` docs sync, right?

Comment (z3z1ma) [reply dbtmetabase/__init__.py:72]: I actually had a bit of confusion there and Im not fully settled on this particular change. But I'll offer what I know and curious as to your comments.

Schema is a required arg on `export_models` function as well as the `build_metadata_lookups`. In that function, the `schema` arg restricts documentation/foriegn_key/semantic_type propagation outwards to only Metabase tables that matched the passed schema. Author likely intended separate invocations per schema? This is independent of the dbt folder / manifest reader and I believe this was in place prior to our work. So if you ran it on your fork without passing schema (took another look at the fork to be sure), a string definitely needed to be passed to `export` and to `build_metadata_lookups`. 

So my question is it seems we have sort of two different intentions running side by side here. If someone can clarify the best case desired behavior and can correct (in the metabase client would be the place most likely).

**Comment by z3z1ma** [comment]
Working on documentation updates now
Then will move onto some pretty cool unit tests

Code is in a reviewable state as is though

**Comment by z3z1ma** [comment]
image

image


Example Native query documentation parsed from metabase into exposure with verbose documentation auto-generated.


image

Example DAG of exposures on a single focused node.

**Pull request closed without merging**